src/Uno.UWP/ApplicationModel/DataTransfer/DataPackage.cs

```diff
@@ -93,6 +93,154 @@ public DataPackageView GetView()
+		/// Null will be returned if the URI/URL is not in the <see cref="StandardDataFormats.WebLink"/> format.
+		/// </param>
+		/// <param name="applicationLink">
+		/// The <see cref="StandardDataFormats.ApplicationLink"/> format Uri intended for use in a <see cref="DataPackage"/>.
+		/// Null will be returned if the URI/URL is not in the <see cref="StandardDataFormats.ApplicationLink"/> format.
+		/// </param>
+		internal static void SeparateUri(string? uri, out string? webLink, out string? applicationLink)
+		{
+			/* UWP has the following standard data formats that correspond with a URI/URL:
+			 *
+			 *  1. Uri, now deprecated in favor of:
+			 *  2. WebLink and
+			 *  3. ApplicationLink
+			 *
+			 * Several platforms such as macOS/iOS/Android do not differentiate between them and
+			 * only use a single URI/URL class or string.
+			 * 
+			 * Therefore, this method is used to map between a native URI/URL and the equivalent
+			 * UWP data type (since UWP's direct equivalent standard data format 'Uri' is deprecated).
+			 * 
+			 * The mapping is as follows:
+			 * 
+			 * 1. WebLink is used if the given URL/URI has a scheme of "http" or "https"
+			 * 2. ApplicationLink is used if not #1
+			 *
+			 * For full compatibility, the Uri format within a DataPackage should still be populated
+			 * regardless of #1 or #2.
+			 */
+
+			uri = uri?.Trim();
+
+			if (string.IsNullOrEmpty(uri))
+			{
+				webLink = null;
+				applicationLink = null;
+			}
+			else if (IsUriWebLink(uri))
+			{
+				webLink = uri;
+				applicationLink = null;
+			}
+			else
+			{
+				webLink = null;
+				applicationLink = uri;
+			}
+
+			return;
+		}
+
+		/// <summary>
+		/// Combines separate URI's for WebLink, ApplicationLink and Uri (deprecated) into a single URI/URL.
+		/// This is useful for converting formats supported by the <see cref="DataPackage"/> into a platform-specific URI/URL.
+		/// </summary>
+		/// <remarks>
+		/// This method is intended for use during integration with other platforms.
+		/// Therefore, it uses strings as the intermediate type instead of a Uri or native URL class.
+		/// </remarks>
+		/// <param name="webLink">
+		/// Data from a <see cref="DataPackage"/> for the <see cref="StandardDataFormats.WebLink"/> format.
+		/// Set to null if this format does not exist in the package.
+		/// </param>
+		/// <param name="applicationLink">
+		/// Data from a <see cref="DataPackage"/> for the <see cref="StandardDataFormats.ApplicationLink"/> format.
+		/// Set to null if this format does not exist in the package.
+		/// </param>
+		/// <param name="uri">
+		/// Data from a <see cref="DataPackage"/> for the <see cref="StandardDataFormats.Uri"/> format.
+		/// Set to null if this format does not exist in the package.
+		/// </param>
+		/// <returns></returns>
+		internal static string CombineUri(string? webLink, string? applicationLink, string? uri)
```

Comment:
Codacy found an issue: [Change this return type to 'System.Uri'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6364607)